src/catalua.cpp

```diff
@@ -824,9 +824,90 @@ int call_lua(std::string tocall)
     return err;
 }
 
-void lua_callback(const char *callback_name)
+void lua_delete_global( const char* name ) {
+    lua_State *L = lua_state;
+    lua_pushnil( L );
+    lua_setglobal( L, name );
+}
+
+void lua_callback_cleanup()
+{
+    lua_delete_global( "callback_last" );
+    lua_delete_global( "callback_arg_count" );
+    lua_delete_global( "mapgen_generator_type" );
+    lua_delete_global( "mapgen_terrain_type_id" );
+    lua_delete_global( "mapgen_terrain_coordinates" );
+    lua_delete_global( "skill_increased_source" );
+    lua_delete_global( "skill_increased_id" );
+    lua_delete_global( "skill_increased_level" );
+    lua_delete_global( "mutation_gained" );
+    lua_delete_global( "mutation_lost" );
+    lua_delete_global( "stat_changed" );
+    lua_delete_global( "stat_value" );
```

Comment:
This looks wrong. Why should there be a global list of *all* possible parameters to the callback functions? What if someone adds a new callback and forgets to add the parameter name here?

Your `CallbackArgumentContainer` type already knows the names of all parameters, so it can be used to remove only those variables in Lua that were actually set up.

Reply:
Actually, I want to clean existing callback arguments whenever new callback is raised. Maybe I should simply create a table in `_G` named `callback_arguments`, store all callback arguments there and clean this table?